urllib3/contrib/gaepool.py

```diff
@@ -0,0 +1,101 @@
+import warnings
+
+from ..connectionpool import (
+    ConnectionPool,
+    _Default,
+    get_host)
+
+from ..exceptions import (
+    HTTPWarning,
+    TimeoutError,
+)
+
+from ..packages.six import BytesIO
+from ..request import RequestMethods
+from ..response import HTTPResponse
+from ..util.timeout import Timeout
+from ..util.url import Url
+
+try:
+    from google.appengine.api import urlfetch
+except ImportError:
+    urlfetch = None
+
+
+class GAEPlatformWarning(HTTPWarning):
+    pass
+
+
+class GAEConnectionPool(ConnectionPool, RequestMethods):
+
+    def __init__(self, host, port=None, scheme='http', timeout=5, headers=None,
+                 retries=None, validate_certificate=False):
```

Comment:
About `validate_certificate`:

* Do we want to default to `False`?
   For our normal connectionpool we have to as we don't have a cert bundle.
   Recently we began defaulting to `True` as soon as we get a bundle.
* Should we emit a warning when `validate_certficate=True and scheme != 'https'`?